airbyte-integrations/connectors/destination-jdbc/src/main/java/io/airbyte/integrations/destination/jdbc/copy/azure/AzureBlobStorageStreamCopier.java

```diff
@@ -0,0 +1,279 @@
+
+  }
+
+  @Override
+  public void closeStagingUploader(boolean hasFailed) throws Exception {
+    LOGGER.info("Uploading remaining data for {} stream.", streamName);
+    for (final var csvPrinter : csvPrinters.values()) {
+      csvPrinter.close();
+    }
+    LOGGER.info("All data for {} stream uploaded.", streamName);
+  }
+
+  @Override
+  public void createDestinationSchema() throws Exception {
+    LOGGER.info("Creating schema in destination if it doesn't exist: {}", schemaName);
+    sqlOperations.createSchemaIfNotExists(db, schemaName);
+  }
+
+  @Override
+  public void createTemporaryTable() throws Exception {
+    LOGGER.info("Preparing tmp table in destination for stream: {}, schema: {}, tmp table name: {}.", streamName, schemaName, tmpTableName);
+    sqlOperations.createTableIfNotExists(db, schemaName, tmpTableName);
+  }
+
+  @Override
+  public void copyStagingFileToTemporaryTable() throws Exception {
+    LOGGER.info("Starting copy to tmp table: {} in destination for stream: {}, schema: {}.", tmpTableName, streamName, schemaName);
+    for (final var azureStagingFile : azureStagingFiles) {
+      copyAzureBlobCsvFileIntoTable(db, getFullAzurePath(azureStagingFile), schemaName, tmpTableName, azureBlobConfig);
+    }
+    LOGGER.info("Copy to tmp table {} in destination for stream {} complete.", tmpTableName, streamName);
+  }
+
+  private String getFullAzurePath(String azureStagingFile) {
+    return "azure://" + azureBlobConfig.getAccountName() + "." + azureBlobConfig.getEndpointDomainName()
+        + "/" + azureBlobConfig.getContainerName() + "/" + azureStagingFile;
+  }
+
+  @Override
+  public String createDestinationTable() throws Exception {
+    var destTableName = nameTransformer.getRawTableName(streamName);
+    LOGGER.info("Preparing table {} in destination.", destTableName);
+    sqlOperations.createTableIfNotExists(db, schemaName, destTableName);
+    LOGGER.info("Table {} in destination prepared.", tmpTableName);
+
+    return destTableName;
+  }
+
+  @Override
+  public String generateMergeStatement(String destTableName) throws Exception {
+    LOGGER.info("Preparing to merge tmp table {} to dest table: {}, schema: {}, in destination.", tmpTableName, destTableName, schemaName);
+    var queries = new StringBuilder();
+    if (destSyncMode.equals(DestinationSyncMode.OVERWRITE)) {
+      queries.append(sqlOperations.truncateTableQuery(db, schemaName, destTableName));
+      LOGGER.info("Destination OVERWRITE mode detected. Dest table: {}, schema: {}, truncated.", destTableName, schemaName);
+    }
+    queries.append(sqlOperations.copyTableQuery(db, schemaName, tmpTableName, destTableName));
+    return queries.toString();
+  }
+
+  @Override
+  public void removeFileAndDropTmpTable() throws Exception {
+    LOGGER.info("Begin cleaning azure blob staging files.");
+    for (AppendBlobClient appendBlobClient : blobClients.values()) {
+      if (appendBlobClient.exists()) {
```

Comment:
does it make sense to check if the file exists before deleting it?

Reply:
removed redundant check if blob exists